<h2 align='center' style='color:blue'>Finding best model and hyper parameters for sklearn wine dataset classification

In [1]:
from sklearn.datasets import load_wine

wine = load_wine()

In [6]:
import pandas as pd
import numpy as np
from sklearn import svm
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import GaussianNB
from sklearn.naive_bayes import MultinomialNB
from sklearn.tree import DecisionTreeClassifier
from sklearn.linear_model import LinearRegression


model_params = {
    'svm':{
        'model':svm.SVC(gamma='auto'),
        'params': {
            'C':[1,15,30],
            'kernel':['rbf','linear'],
        }
    },
    'rand_f':{
        'model':RandomForestClassifier(),
        'params':{
            'n_estimators':[1,10,20],
        }
    },
    'log_reg':{
        'model':LogisticRegression(solver='liblinear',multi_class='auto'),
        'params': {
            'C':[1,5,10],
        }
    },
    'G_nb':{
        'model':GaussianNB(),
        'params':{},
    },
    'M_nb':{
        'model':MultinomialNB(),
        'params':{}
    },
    'lin_reg':{
        'model':LinearRegression(),
        'params':{}
    },
    'De_tree':{
        'model':DecisionTreeClassifier(),
        'params':{
            'criterion':['gini','entropy'],
        }
    },

}

In [9]:
from sklearn.model_selection import GridSearchCV

scores = []
for model_names, mp in model_params.items():
    clf = GridSearchCV(mp['model'], mp['params'], cv=5, return_train_score=False)
    clf.fit(wine.data,wine.target),
    scores.append({
        'model':model_names,
        'best_score':clf.best_score_,
        'best_params':clf.best_params_
    })

In [7]:
df = pd.DataFrame(scores)
df

model  best_score                   best_params
0      svm    0.961111  {'C': 1, 'kernel': 'linear'}
1   rand_f    0.977778          {'n_estimators': 10}
2  log_reg    0.961111                      {'C': 1}
3     G_nb    0.966349                            {}
4     M_nb    0.843968                            {}
5  lin_reg    0.183425                            {}
6  De_tree    0.899206      {'criterion': 'entropy'}

**The best model that can be used for better score is RandomForestClassifier with a 97% score accuracy.**